code/modules/surgery/organs/augments_arms.dm

```diff
@@ -559,3 +559,35 @@
 		qdel(src)
 		qdel(I)
 	return ..()
+
+/obj/item/organ/internal/cyberimp/arm/muscle
+	name = "Strong-arm empowered musculature implant"
+	desc = "When implanted, this cybernetic implant will enhance the muscles of the arm to deliver more power-per-action. Only has to be installed in one arm."
+	icon_state = "muscle_imp"
+
+	parent_organ = "l_arm" //Left arm by default
+	slot = "l_arm_device"
+
+	actions_types = list()
+
+/obj/item/organ/internal/cyberimp/arm/muscle/insert(mob/living/carbon/M, special, dont_remove_slot)
+	. = ..()
+	var/datum/martial_art/muscle_implant/muscle_implant = new(null)
+	muscle_implant.teach(M)
+
+/obj/item/organ/internal/cyberimp/arm/muscle/remove(mob/living/carbon/M, special)
+	. = ..()
+	var/datum/martial_art/muscle_implant/muscle_implant = new(null)
+	muscle_implant.remove(M)
+
+/obj/item/organ/internal/cyberimp/arm/muscle/emp_act(severity)
+	. = ..()
+	if(emp_proof)
+		return
+	to_chat(owner, "<span class='danger'>Your arm spasms wildly!</span>")
+	ADD_TRAIT(owner, MUSCLE_SPASMS, "muscle_imp_EMP")
+	addtimer(CALLBACK(src, PROC_REF(reboot)), (18 / severity) SECONDS)
+
+/obj/item/organ/internal/cyberimp/arm/muscle/proc/reboot()
+	REMOVE_TRAIT(owner, MUSCLE_SPASMS, "muscle_imp_EMP")
+	to_chat(owner, "<span class='danger'>Your arms stop spasming.")
```

Comment:
It also means that this trait can be removed and made a proc on `/datum/martial_art/muscle_implant`

Reply:
How would you suggest doing this? `emp_act()` isn't defined on the datum

Reply:
`/datum/martial_art/muscle_implant/proc/on_emp(severity)` and just call that after emp_proof